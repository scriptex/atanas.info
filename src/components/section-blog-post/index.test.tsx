import * as React from 'react';
import { render } from 'enzyme';

import { SectionBlogPost } from '.';

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useParams: jest.fn().mockReturnValue({ slug: 'bulk-image-watermark-with-nodejs-and-jimp' })
}));

jest.mock('../../data/articles', () => ({
	articles: [
		{
			url: '/blog/bulk-image-watermark-with-nodejs-and-jimp',
			title: 'Bulk image watermark with NodeJS and Jimp',
			image: '/images/temp/bulk-image-watermark-with-nodejs-and-jimp.jpg',
			content: jest.fn(() => <div>Content</div>),
			external: false
		}
	]
}));

describe('SectionBlogPost component', () => {
	it('Should render the SectionBlogPost component', () => {
		const wrapper = render(<SectionBlogPost />);

		expect(wrapper).toMatchSnapshot();
	});
});
