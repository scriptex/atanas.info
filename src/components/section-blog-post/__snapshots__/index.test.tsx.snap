// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionBlogPost component Should render the SectionBlogPost component 1`] = `
<main
  class="o-main "
>
  <section
    class="c-section c-section--blog-post"
    id="blog-post"
  >
    <div
      class="o-shell"
    >
      <div
        class="lines"
      >
        <div
          class="line line--vertical"
        />
        <div
          class="line line--vertical"
        />
        <div
          class="line line--vertical"
        />
      </div>
      <div
        class="lines"
      >
        <div
          class="line line--horizontal"
        />
        <div
          class="line line--horizontal"
        />
        <div
          class="line line--horizontal"
        />
      </div>
      <h1>
        Bulk image watermark with NodeJS and Jimp
      </h1>
      <img
        alt=""
        src="/images/temp/bulk-image-watermark-with-nodejs-and-jimp.jpg"
      />
      <div
        class="c-blog-post"
      >
        <div>
          Content
        </div>
      </div>
      <div
        class="c-section__actions"
      >
        <button
          class="c-btn "
          type="button"
        >
          Get in touch
        </button>
        <ul
          class="c-section__actions-list"
        >
          <li>
            <button
              class="c-btn c-section__actions-close"
              type="button"
            >
              Close
            </button>
          </li>
          <li>
            <button
              class="c-btn "
              type="button"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  href="#svg-email"
                />
              </svg>
              <span>
                Contact me
              </span>
              <strong>
                directly
              </strong>
            </button>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://app.usebraintrust.com/talent/782/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  href="#svg-braintrust"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Braintrust
              </strong>
            </a>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://www.toptal.com/resume/atanas-atanasov"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  href="#svg-toptal"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Toptal
              </strong>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </section>
</main>
`;
